extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -507,10 +510,12 @@ export class AmpConsent extends AMP.BaseElement {
     const href =
         this.consentConfig_[instanceId]['checkConsentHref'];
     assertHttpsUrl(href, this.element);
-    const viewer = Services.viewerForDoc(this.getAmpDoc());
+    const ampDoc = this.getAmpDoc();
+    const resolvedHref = resolveRelativeUrl(href, ampDoc.getUrl());
```

Comment:
This is incorrect, it needs to be resolved from the `sourceUrl`, not the ampdoc's url.